hal/src/main/native/systemcore/FRCDriverStation.cpp

```diff
@@ -432,6 +433,15 @@ int32_t HAL_GetControlWord(HAL_ControlWord* controlWord) {
   return 0;
 }
 
+int32_t HAL_GetOpMode(char* buf, int32_t* len) {
+  std::scoped_lock lock{cacheMutex};
+  *len = std::min(static_cast<size_t>(*len), currentRead->opMode.length());
+  if (*len != 0) {
+    std::memcpy(buf, currentRead->opMode.data(), *len);
+  }
+  return 0;
```

Comment:
I feel like truncating here is a bad idea, and instead we should return an error if it doesn't fit.

Reply:
I committed a WIP because I was switching machines, this is still the old string-based implementation for GetOpMode.